BVH.cs

```diff
@@ -202,6 +236,26 @@ public void GetAllNodeMatriciesRecursive(BVHNode<T> n, ref List<Matrix4x4> matri
             if (n.Left != null) GetAllNodeMatriciesRecursive(n.Left, ref matricies, depth + 1);
         }
 
+		public Matrix4x4 GetNodeMatrix(BVHNode<T> node) => Matrix4x4.Translate(node.Box.center) * Matrix4x4.Scale(node.Box.size);
+		
+
+		public void RenderNode(BVHNode<T> node) {
+			Matrix4x4 mat = GetNodeMatrix(node);
+			List<Matrix4x4> matrices = new List<Matrix4x4>();
+			matrices.Add(mat);
+			Mesh mesh = new Mesh();
+            mesh.SetVertices(vertices);
+            mesh.SetIndices(indices, MeshTopology.Lines, 0);
+			if(_debugRenderMaterial == null)
+			{
+				_debugRenderMaterial = new Material(Shader.Find("Standard"))
```

Comment:
Is this compatible with SRP?

Reply:
I will test this to make sure it is and will come back to you :)